src/Sdk/DTWebApi/WebApi/Runner.cs

```diff
@@ -1,18 +1,73 @@
+using System;
 using Newtonsoft.Json;
 
 namespace GitHub.DistributedTask.WebApi
 {
     public class Runner
     {
-        /// <summary>
-        /// Name of the agent
-        /// </summary>
+
+        public class Authorization
+        {
+            /// <summary>
+            /// The url to refresh tokens
+            /// </summary> 
+            [JsonProperty("authorization_url")]
+            public Uri AuthorizationUrl
+            {
+                get;
+                internal set;
+            }
+
+            /// <summary>
+            /// The url to connect to to poll for messages
+            /// </summary> 
+            [JsonProperty("server_url")]
+            public string ServerUrl
+            {
+                get;
+                internal set;
+            }
+
+            /// <summary>
+            /// The client id to use when connecting to the authorization_url
+            /// </summary>
+            [JsonProperty("client_id")]
+            public string ClientId
+            {
+                get;
+                internal set;
+            }
+        }
+
         [JsonProperty("name")]
         public string Name
         {
             get;
             internal set;
         }
 
+        [JsonProperty("id")]
+        public Int32 Id
+        {
+            get;
+            internal set;
+        }
+
+        [JsonProperty("authorization")]
+        public Authorization RunnerAuthorization
+        {
+            get;
+            internal set;
+        }
```

Comment:
Do we need to mark these new fields as optional in order to support all rollback and rollforward scenarios?

Reply:
They should be optional by default: <img width="810" alt="Screenshot 2023-03-27 at 11 22 49 AM" src="https://user-images.githubusercontent.com/19689611/228048697-fb537e5e-f869-4062-8ce1-2e375937a8e6.png">